### PR TITLE
libcvmfs: make modifications for js-monitor

### PR DESCRIPTION
**Pull request by janpriessnitz**

### CODE REVIEW

File: cvmfs/libcvmfs.h [hunk 1]
```diff
@@ -429,6 +453,13 @@ int cvmfs_listdir_contents(
   size_t *listlen,
   size_t *buflen);
 
+int cvmfs_listdir_stat(
```
Comment (jblomer) [inline cvmfs/libcvmfs.h:456]: Add a comment

File: cvmfs/libcvmfs_int.cc [hunk 2]
```diff
@@ -470,10 +529,28 @@ int LibContext::GetNestedCatalogAttr(
...
+  std::string subcat_path;
+  std::map<std::string, uint64_t> counters =
+    mount_point_->catalog_mgr()->LookupCounters(p, &subcat_path).GetValues();
+
   // Set values of the passed structure
   nc_attr->mountpoint = strdup(mountpoint.ToString().c_str());
   nc_attr->hash = strdup(hash.ToString().c_str());
   nc_attr->size = size;
+
+
```
Comment (jblomer) [inline cvmfs/libcvmfs_int.cc:541]: Remove one blank line

File: cvmfs/libcvmfs.h (hunk 1 above)
Comment (jblomer) [inline cvmfs/libcvmfs.h:456]: Let's add a unit test for the new routine

**Comment by janpriessnitz** [comment]
@jblomer I've sorted out the comments, so everything should be ready. However, let me test it for one more time with cvmfs-monitor-backend before merging.

**Comment by janpriessnitz** [comment]
@cernvm-bot unittest

**Comment by janpriessnitz** [comment]
@cernvm-bot unittest

**Comment by janpriessnitz** [comment]
Alright, works fine. @jblomer please merge.

**Comment by jblomer** [comment]
@cernvm-bot cloudtest

**Pull request merged**